### ISSUE TITLE
Can't remove artifact linked to ACR from Release Definition

### ISSUE THREAD

**Issue by ppanyukov** [Microsoft/vsts-agent#1387]
## VSTS type and version
VisualStudio.com

## What's not working?
We have one release definition which has artifact linked to Azure Container Registry.  And we have Continuous Devliver optioned switched on.

Now when we try to switch off CD, or remove the artifact from the release definition and then try to save it, we get an error "Object reference not set to an instance of an object.".  Same happens if we try to delete the definition.

Basically we are stuck.  Any thoughts how we may resolve this? 

<img width="564" alt="screen shot 2018-01-29 at 18 14 11" src="https://user-images.githubusercontent.com/300031/35526762-6cbe7a0a-0520-11e8-9955-f448788c12c5.png">

**Comment by TingluoHuang** [comment]
@GitHubSriramB

**Comment by TingluoHuang** [comment]
@ppanyukov please use https://developercommunity.visualstudio.com/spaces/21/index.html to report this kind of issue, since the issue you reported is out of the scope for this repository.
It's not an issue with the build/release agent.

**Comment by TingluoHuang** [comment]
I am going to close this issue after @GitHubSriramB acknowledged the problem in Release.

**Comment by GitHubSriramB** [comment]
@ppanyukov - Apologies for the inconvenience. We will investigate and give an update.

**Comment by jikuma** [comment]
Is your account name talktalktv.visualstudio.com? This is a known issue and it happens when the endpoint used in the artifact is deleted. Have you deleted the endpoint used in the ACR artifact.

**Comment by ppanyukov** [comment]
@jikuma @GitHubSriramB Yes, account is `talktalktv.visualstudio.com`.

I've also raised this issue here (as per advice): https://developercommunity.visualstudio.com/content/problem/190983/cant-remove-artifact-linked-to-acr-from-release-de.html

Some additional detail from there:

> Using the devtools, we are getting HTTP 500:
>
> - when trying to delete the definition:
>
> `DELETE https://[redacted].vsrm.visualstudio.com/TV/_apis/Release/definitions/80?forceDelete=false`
>
> - when trying to save the definition:
>
> `PUT https://[redacted].vsrm.visualstudio.com/TV/_apis/Release/definitions`
>
> In both cases the body of the response has this:
>
> ```
> {
> "$id": "1",
> "innerException": null,
> "message": "Object reference not set to an instance of an object.",
> "typeName": "Microsoft.TeamFoundation.Framework.Server.ActionDeniedBySubscriberException, > Microsoft.TeamFoundation.Framework.Server",
> "typeKey": "ActionDeniedBySubscriberException",
> "errorCode": 0,
> "eventId": 4223
> }
> ```

Sorry for all this double-posting! The boundaries of support are a bit confusing when it comes to VSTS.

**Comment by ppanyukov** [comment]
@jikuma It's a bit more complicated than just deleting the endpoint.

All endpoints were there when I first tried deleting the release definition and if failed. I then noticed there as a "web hooks" resource in Azure Portal, in the same resource group as the ACR. There were quite a few hooks there. 

I then tried to delete this resource, and the portal wouldn't let me saying "There is a lock on the *resource group*". Which is odd. Yes, there was a delete lock on the RG, but not on the webhooks reource itself. 

I *think* I tried to delete the release definition after I removed the lock on RG. If I did, then it didn't work.

After that I deleted the entire webhooks object. And still it didn't work.

I'm pretty certain the actual ACR stayed the same all the time (wasn't deleted or anything).

So at the moment we still have the ACR, but don't have the webhooks. There are no locks on the RG or any other resources in that RG.
